frontend/packages/dev-console/src/components/helm/form/HelmInstallUpgradeForm.tsx

```diff
@@ -19,37 +19,42 @@ const HelmInstallUpgradeForm: React.FC<FormikProps<FormikValues> & HelmInstallUp
   handleReset,
   status,
   isSubmitting,
-  submitLabel,
+  helmAction,
   values,
   dirty,
 }) => {
   const { chartName, chartVersion } = values;
+  const isSubmitDisabled =
+    helmAction !== HelmActionType.Install &&
+    ((chartVersion && !dirty) || status?.isSubmitting || !_.isEmpty(errors));
   return (
     <FlexForm onSubmit={handleSubmit}>
       <FormSection fullWidth>
         <Grid gutter={'md'}>
-          <GridItem span={submitLabel === HelmActionType.Install ? 12 : 6}>
+          <GridItem span={6}>
             <InputField
               type={TextInputTypes.text}
               name="helmReleaseName"
               label="Release Name"
               helpText="A unique name for the Helm Chart release."
               required
-              isDisabled={!!chartVersion}
+              isDisabled={helmAction === HelmActionType.Upgrade}
             />
           </GridItem>
-          {chartVersion && (
-            <HelmChartVersionDropdown chartName={chartName} chartVersion={chartVersion} />
-          )}
+          <HelmChartVersionDropdown
+            chartName={chartName}
+            chartVersion={chartVersion}
+            helmAction={helmAction}
+          />
         </Grid>
       </FormSection>
       {chartHasValues && <YAMLEditorField name="chartValuesYAML" onSave={handleSubmit} />}
       <FormFooter
         handleReset={handleReset}
         errorMessage={status && status.submitError}
         isSubmitting={status?.isSubmitting || isSubmitting}
-        submitLabel={submitLabel}
-        disableSubmit={(chartVersion && !dirty) || status?.isSubmitting || !_.isEmpty(errors)}
+        submitLabel={helmAction}
+        disableSubmit={isSubmitDisabled}
```

Comment:
The submit button is not getting disabled now while submitting the form.

Reply:
fixed